src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,383 @@
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.validation.TransactionSolidifier;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.log.interval.IntervalLogger;
+import com.iota.iri.utils.thread.DedicatedScheduledExecutorService;
+import com.iota.iri.utils.thread.SilentScheduledExecutorService;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+import static com.iota.iri.controllers.TransactionViewModel.PREFILLED_SLOT;
+import static com.iota.iri.controllers.TransactionViewModel.fromHash;
+
+/**
+ * A solidifier class for processing transactions. Transactions are checked for solidity, and missing transactions are
+ * subsequently requested. Once a transaction is solidified correctly it is placed into a broadcasting set to be sent to
+ * neighboring nodes.
+ */
+public class TransactionSolidifierImpl implements TransactionSolidifier {
+
+    private Tangle tangle;
+    private SnapshotProvider snapshotProvider;
+    private TransactionRequester transactionRequester;
+
+    /**
+     * Max size for all queues.
+     */
+    private static final int MAX_SIZE= 10000;
+
+    private static final int SOLIDIFICATION_INTERVAL = 100;
+
+    private static final IntervalLogger log = new IntervalLogger(TransactionSolidifier.class);
+
+    /**
+     * Executor service for running the {@link #processTransactionsToSolidify()}.
+     */
+    private SilentScheduledExecutorService executorService = new DedicatedScheduledExecutorService(
+            "Transaction Solidifier", log.delegate());
+
+    /**
+     * A queue for processing transactions with the {@link #checkSolidity(Hash)} call. Once a transaction has been
+     * marked solid it will be placed into the {@link #transactionsToBroadcast} queue.
+     */
+    private BlockingQueue<Hash> transactionsToSolidify = new ArrayBlockingQueue<>(MAX_SIZE);
+
+    /**
+     * A queue for processing transactions with the {@link #propagateSolidTransactions()} call. This will check
+     * approving transactions with {@link #quickSetSolid(TransactionViewModel)}.
+     */
+    private BlockingQueue<Hash> solidTransactions = new ArrayBlockingQueue<>(MAX_SIZE);
+
+    /**
+     * A set of transactions that will be called by the {@link TransactionProcessingPipeline} to be broadcast to
+     * neighboring nodes.
+     */
+    private BlockingQueue<TransactionViewModel> transactionsToBroadcast = new ArrayBlockingQueue<>(MAX_SIZE);
+
+    private TipsViewModel tipsViewModel;
+
+    /**
+     * Constructor for the solidifier.
+     * @param tangle                    The DB reference
+     * @param snapshotProvider          For fetching entry points for solidity checks
+     * @param transactionRequester      A requester for missing transactions
+     */
+    public TransactionSolidifierImpl(Tangle tangle, SnapshotProvider snapshotProvider, TransactionRequester transactionRequester,
```

Comment:
Maybe have an inner class for the propagation solidifier?
just to separate between responsibilities?